strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAPartitionedEntityManagerPool.java

```diff
@@ -0,0 +1,109 @@
+package com.orientechnologies.orient.object.jpa;
+
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.persistence.Cache;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnitUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.metamodel.Metamodel;
+
+import com.orientechnologies.orient.core.db.OPartitionedDatabasePool;
+import com.orientechnologies.orient.object.db.OObjectDatabaseTx;
+
+/**
+ * @author Sergey Bespalov
+ *
+ */
+public class OJPAPartitionedEntityManagerPool implements EntityManagerFactory
```

Comment:
Do not worry, I plan to contribute this code to OrientDB, and if they will accept this, we will be able to remove this with future OrientDB releases. 
The goal of this, and main difference with current `OJPAEntityManagerFactory`, is that it based on `OPartitionedDatabasePool`, which is the OrientDB connection pool.